assistants/models.tsp

```diff
@@ -215,24 +332,68 @@ model AssistantObject {
    */
   @extension("x-oaiTypeLabel", "map")
   metadata: Record<string> | null;
+
+  /**
+   * What sampling temperature to use, between 0 and 2. Higher values like 0.8 will make the output
+   * more random, while lower values like 0.2 will make it more focused and deterministic.
+   */
+  @minValue(0)
+  @maxValue(2)
+  temperature?: float64 | null = 1;
+
+  /**
+   * An alternative to sampling with temperature, called nucleus sampling, where the model
+   * considers the results of the tokens with top_p probability mass. So 0.1 means only the tokens
+   * comprising the top 10% probability mass are considered.
+   * 
+   * We generally recommend altering this or temperature but not both.
+   */
+  @minValue(0)
+  @maxValue(1)
+  top_p?: float64 | null = 1;
+
+  response_format?: AssistantsApiResponseFormatOption | null;
 }
 
-/** A list of [Files](/docs/api-reference/files) attached to an `assistant`. */
-model AssistantFileObject {
-  /** The identifier, which can be referenced in API endpoints. */
-  id: string;
+/**
+ * Controls for how a thread will be truncated prior to the run. Use this to control the intial
+ * context window of the run.
+ */
+model TruncationObject {
+  /**
+   * The truncation strategy to use for the thread. The default is `auto`. If set to
+   * `last_messages`, the thread will be truncated to the n most recent messages in the thread.
+   * When set to `auto`, messages in the middle of the thread will be dropped to fit the context
+   * length of the model, `max_prompt_tokens`.
+   */
+  type: "auto" | "last_messages";
 
-  /** The object type, which is always `assistant.file`. */
-  object: "assistant.file";
+  /**
+   * The number of most recent messages from the thread when constructing the context for the run.
+   */
+  @minValue(1)
+  last_messages?: int32 | null;
+}
 
-  /** The Unix timestamp (in seconds) for when the assistant file was created. */
-  @encode("unixTimestamp", int32)
-  @extension(
-    "x-customTspIngestionRule",
-    "_at and created are Unix encoded utcDateTime"
-  )
-  created_at: utcDateTime;
+/**
+ * Controls which (if any) tool is called by the model.
+ * `none` means the model will not call any tools and instead generates a message.
+ * `auto` is the default value and means the model can pick between generating a message or calling
+ * one or more tools.
+ * `required` means the model must call one or more tools before responding to the user.
+ * Specifying a particular tool like `{"type": "file_search"}` or `{"type": "function", "function":
+ * {"name": "my_function"}}` forces the model to call that tool.
+ */
+alias AssistantsApiToolChoiceOption = "none" | "auto" | "required" | AssistantsNamedToolChoice;
```

Comment:
```suggestion
@extension("x-oaiExpandable", true)
alias AssistantsApiToolChoiceOption = "none" | "auto" | "required" | AssistantsNamedToolChoice;
```

![image](https://github.com/joseharriaga/openai-in-typespec/assets/21109913/6df85084-03d7-405a-84bd-58da8af0bf2a)